### PR TITLE
DIGI-13286, DIGI-13289, DIGI-13290 - Filter out specific values into the Session dictionary

### PR DESCRIPTION
**Pull request by hamiltonalex**

### CODE REVIEW

File: DigiMeSDK/Core/Classes/Entities/CASession.m [hunk 1]
```diff
@@ -11,6 +11,39 @@
...
+NSString * const kDebugBundleVersion = @"debugBundleVersion";
+NSString * const kDebugPlatform = @"debugPlatform";
+NSString * const kContractType = @"debugContractType";
+NSString * const kDeviceId = @"debugDeviceId";
+NSString * const kDigiMeVersion = @"debugDigiMeVersion";
+NSString * const kUserId = @"debugUserId";
+NSString * const kLibraryId = @"debugLibraryId";
+NSString * const kPCloudType = @"debugPCloudType";
+
+@interface CASession()
```
[the remark this answers is not in the document]
Comment (michaeleustace) [reply DigiMeSDK/Core/Classes/Entities/CASession.m]: The private header should only expose properties that are already in the .m file. A property which is declared in a category cannot then be implemented in another @impplementation of that class (e.g. in CASeesion+Private.m.

File: DigiMeSDK/Core/Classes/DMEAuthorizationManager.m [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 #import "DMEAuthorizationManager.h"
 #import "CASessionManager.h"
+#import "CASession.h"
```
Comment (brokenzz) [inline DigiMeSDK/Core/Classes/DMEAuthorizationManager.m]: You probably don't need this here, as it is imported in the private one :)

File: DigiMeSDK/Core/Classes/Entities/CASession.h [hunk 3]
```diff
@@ -68,6 +94,12 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, strong, readonly, nullable) id<CADataRequest> scope;
 
+
+/**
+ Metadata - exposing extra return parameters.
```
Comment (brokenzz) [inline DigiMeSDK/Core/Classes/Entities/CASession.h]: I would reword this comment to something like:

```Objc
Session metadata. Contains additional debug information collected during the session lifetime.
```
